modules/exploits/multi/http/wp_givewp_rce.rb

```diff
@@ -67,12 +74,26 @@ def initialize(info = {})
   def check
     return CheckCode::Unknown unless wordpress_and_online?
 
+    vuln_version_1 = Rex::Version.new('3.14.2')
+
     print_status("WordPress Version: #{wordpress_version}") if wordpress_version
-    check_code = check_plugin_version_from_readme('give', '3.14.2')
+
+    check_code = check_plugin_version_from_readme('give')
     return CheckCode::Safe unless check_code.code == 'appears'
 
-    print_good("Detected GiveWP Plugin version: #{check_code.details[:version]}")
-    CheckCode::Appears
+    detected_version = Rex::Version.new(check_code.details[:version])
+    print_good("Detected GiveWP Plugin version: #{detected_version}")
+
+    if detected_version <= vuln_version_1
+      print_good('Vulnerable to both CVE-2024-5932 and CVE-2024-8353 (bypass).')
+      return CheckCode::Appears
+    elsif detected_version > vuln_version_1
+      print_good('Vulnerable to CVE-2024-8353 (bypass).')
+      return CheckCode::Appears
+    end
+
+    print_status("GiveWP Plugin version #{detected_version} is not vulnerable.")
```

Comment:
I think the logic is wrong here, as the two branches of the `if` cover all cases.

Reply:
Yeah actually I had made another check function I don't know what I did with my commits. I'll fix that soon.